src/runtime/internal/utils.ts

```diff
@@ -96,7 +96,9 @@ export let now: () => number = is_client
 	? () => window.performance.now()
 	: () => Date.now();
 
-export let raf = is_client ? requestAnimationFrame : noop;
+export let raf = (callback) => is_client && window.requestAnimationFrame
+	? window.requestAnimationFrame.call(window, callback)
+	: undefined;
```

Comment:
Is there a reason you changed this from `noop` to `undefined` when we're not in the browser with `rAF` available?

Reply:
With the current setup I could either return `undefined`, or call `noop` on this. Both will have the same effect. Personally I think returning `undefined` is clearer, but I can replace with `noop()` instead, no problem. What is your opinion?

Reply:
Hey, sorry about the delay. I forgot that `undefined` was the _return value_ of what's being assigned to `raf`, not what's actually being assigned to it. So what you have is correct.

However, I think I'd prefer to just check _once_ whether we are `is_client`, rather than every time `raf` is called. This:

```js
export let raf = is_client ?
	cb => requestAnimationFrame.call(window, cb) :
	noop;
```

looks to pass all tests locally for me. What do you think about doing that?

Reply:
Actually, won't just `export let raf = is_client ? cb => requestAnimationFrame(cb) : noop;` work? I don't think there's a need to bind if we're specifically calling `requestAnimationFrame` here and not returning it to get called elsewhere.

Reply:
I don't think we even need the `is_client` at that point. `export let raf = cb => requestAnimationFrame(cb);` should be enough.

If someone is calling `raf(cb)` and `requestAnimationFrame` doesn't exist, I'd rather error than silently fail.

Reply:
`cb => requestAnimationFrame(cb)` does pass all the tests, but I would really like to be able to reproduce this issue locally and to see that that fixes it. I don't think it's reasonable to expect our tests to be able to actually test anything here, because they all override rAF anyway with `set_raf`. I tried downloading https://codesandbox.io/s/svelte-raf-invocation-error-yl4or and running that locally, but I no longer saw the exception when I did that.

Reply:
Oh! I wonder if this only happens with webpack or something. There's no scope flattening, and each module gets turned into a separate object, right? Then we call `internal.raf(whatever)` and blam we get the wrong `this`.

Reply:
I can't reproduce it locally with webpack either. But according to the error messages, it looks like codesandbox is using the CJS versions of files. Take a look at https://unpkg.com/svelte@3.4.3/internal and search for `exports.raf(`. That's gonna be the wrong `this`. I wonder whether this is actually revealing a limitation in Rollup's CJS output. I'm not sure what Rollup could be expected to do about this.

Reply:
Ay, tough one!

Although it does seem to be a Rollup limitation in this specific case, it's still unsafe to do this even on esm. `import * as x from 'y'; x.z();` _will_ bind `x` to `z`'s `this`, so we shouldn't ever export functions that rely on `this` anyway.

Good finding!